In [1]:
import pandas as pd
from glob import glob

results_path_TEST = './results_TEST/'

batch_results_TEST = glob(results_path_TEST + '*.csv')

In [2]:
dfs = []
for each_file in batch_results_TEST:
    df = pd.read_csv(each_file)
    dfs.append(df)
batch_results_TEST_df = pd.concat(dfs)

In [3]:
res_df_vis = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_VISUAL', 'Answer.parCoherence_VISUAL',
       'Answer.sentStructure_VISUAL','Answer.wordChoice_VISUAL']]
res_df_vis = res_df_vis[res_df_vis['Input.Model'] == 'VISUAL']

res_df_lang = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_LANGUAGE', 'Answer.parCoherence_LANGUAGE',
       'Answer.sentStructure_LANGUAGE','Answer.wordChoice_LANGUAGE']]
res_df_lang = res_df_lang[res_df_lang['Input.Model'] == 'LANGUAGE']

res_df_vis_lang = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_VIS_LANG', 'Answer.parCoherence_VIS_LANG',
       'Answer.sentStructure_VIS_LANG','Answer.wordChoice_VIS_LANG']]
res_df_vis_lang = res_df_vis_lang[res_df_vis_lang['Input.Model'] == 'VIS_LANG']

res_df_vis_att = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_VISUAL_ATT', 'Answer.parCoherence_VISUAL_ATT',
       'Answer.sentStructure_VISUAL_ATT','Answer.wordChoice_VISUAL_ATT']]
res_df_vis_att = res_df_vis_att[res_df_vis_att['Input.Model'] == 'VISUAL_ATT']

res_df_lang_att = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_LANGUAGE_ATT', 'Answer.parCoherence_LANGUAGE_ATT',
       'Answer.sentStructure_LANGUAGE_ATT','Answer.wordChoice_LANGUAGE_ATT']]
res_df_lang_att = res_df_lang_att[res_df_lang_att['Input.Model'] == 'LANGUAGE_ATT']

res_df_vis_lang_att = batch_results_TEST_df[['Input.Image',
       'Input.Paragraph', 'Input.Model',
       'Answer.objSalience_VIS_LANG_ATT', 'Answer.parCoherence_VIS_LANG_ATT',
       'Answer.sentStructure_VIS_LANG_ATT','Answer.wordChoice_VIS_LANG_ATT']]
res_df_vis_lang_att = res_df_vis_lang_att[res_df_vis_lang_att['Input.Model'] == 'VIS_LANG_ATT']

In [4]:
print(res_df_vis['Answer.wordChoice_VISUAL'].mean())
print(res_df_vis['Answer.objSalience_VISUAL'].mean())
print(res_df_vis['Answer.sentStructure_VISUAL'].mean())
print(res_df_vis['Answer.parCoherence_VISUAL'].mean())

38.12
46.1
42.84
39.06


In [5]:
print(res_df_lang['Answer.wordChoice_LANGUAGE'].mean())
print(res_df_lang['Answer.objSalience_LANGUAGE'].mean())
print(res_df_lang['Answer.sentStructure_LANGUAGE'].mean())
print(res_df_lang['Answer.parCoherence_LANGUAGE'].mean())

34.52
41.38
48.38
40.5


In [6]:
print(res_df_vis_lang['Answer.wordChoice_VIS_LANG'].mean())
print(res_df_vis_lang['Answer.objSalience_VIS_LANG'].mean())
print(res_df_vis_lang['Answer.sentStructure_VIS_LANG'].mean())
print(res_df_vis_lang['Answer.parCoherence_VIS_LANG'].mean())

42.22
48.32
52.88
49.04


In [7]:
print(res_df_vis_att['Answer.wordChoice_VISUAL_ATT'].mean())
print(res_df_vis_att['Answer.objSalience_VISUAL_ATT'].mean())
print(res_df_vis_att['Answer.sentStructure_VISUAL_ATT'].mean())
print(res_df_vis_att['Answer.parCoherence_VISUAL_ATT'].mean())

42.44
37.76
57.84
28.88


In [8]:
print(res_df_lang_att['Answer.wordChoice_LANGUAGE_ATT'].mean())
print(res_df_lang_att['Answer.objSalience_LANGUAGE_ATT'].mean())
print(res_df_lang_att['Answer.sentStructure_LANGUAGE_ATT'].mean())
print(res_df_lang_att['Answer.parCoherence_LANGUAGE_ATT'].mean())

36.68
30.68
49.56
30.5


In [9]:
print(res_df_vis_lang_att['Answer.wordChoice_VIS_LANG_ATT'].mean())
print(res_df_vis_lang_att['Answer.objSalience_VIS_LANG_ATT'].mean())
print(res_df_vis_lang_att['Answer.sentStructure_VIS_LANG_ATT'].mean())
print(res_df_vis_lang_att['Answer.parCoherence_VIS_LANG_ATT'].mean())

38.4
51.28
56.96
35.58


| MODEL | Word Choice | Obj Salience | Sent Structure | Par Coherence |
| --- | --- | --- | --- | --- |
| VISUAL | 38.12 | 46.10 | 42.84 | 39.06 |
| LANGUAGE | 34.52 | 41.38 | 48.38 | <font color='red'>40.50</font> |
| VISUAL LANGUAGE | <font color='red'>42.22</font> | <font color='red'>48.32</font> | 52.88 | <font color='red'>49.04</font> |
| VISUAL ATTENTION | <font color='red'>42.44</font> | 37.76 | <font color='red'>57.84</font> | 28.88 |
| LANGUAGE ATTENTION | 36.68 | 30.68 | 49.56 | 30.50 |
| VISUAL LANGUAGE ATTENTION | 38.4 | <font color='red'>51.28</font> | <font color='red'>56.96</font> | 35.58 |

In [10]:
batch_results_TEST_df.groupby('WorkerId').count()

HITId  HITTypeId  Title  Description  Keywords  Reward  \
WorkerId                                                                 
A130P4M9JPT4YA      5          5      5            5         5       5   
A1CE5G48ELWN0F      1          1      1            1         1       1   
A1EUBMQ86K32XE      2          2      2            2         2       2   
A1GKEEI844CEKI     17         17     17           17        17      17   
A219VCQZADQ45W      2          2      2            2         2       2   
A22VLXXVIEJSD1      1          1      1            1         1       1   
A26UAJWMMNN351      1          1      1            1         1       1   
A27SMEOPKV84VI      3          3      3            3         3       3   
A28L1K6D8QUCML     12         12     12           12        12      12   
A292QSLC0BUT0O      2          2      2            2         2       2   
A29VL3MZE7YPBZ     12         12     12           12        12      12   
A2IQ0QCTQ3KWLT      2          2      2            2         2       2   
A2KLDW5VRXUJ8S      1          1      1            1         1       1   
A2LRFPJOIC51KY     13         13     13           13        13      13   
A2SBN9B6Q3DN2X      1          1      1            1         1       1   
A2WNW8A4MOR7T7      9          9      9            9         9       9   
A31Z5TPD8QKE26     25         25     25           25        25      25   
A36H3PO6JZD414      6          6      6            6         6       6   
A39YBLQYIUBR76     16         16     16           16        16      16   
A3BI0AX5T5GVO3    129        129    129          129       129     129   
A3HMDGRGC5JB7E      1          1      1            1         1       1   
A3MS21UJPMHLK0      1          1      1            1         1       1   
A3OLRWACCCCUTU      1          1      1            1         1       1   
A3V9JFVZQ2XF4Y     20         20     20           20        20      20   
A5V3ZMQI0PU3F       7          7      7            7         7       7   
A8TIL9U6YHG5        2          2      2            2         2       2   
A9HQ3E0F2AGVO       1          1      1            1         1       1   
AHV4U78TUUDKI       5          5      5            5         5       5   
AKSJ3C5O3V9RB       2          2      2            2         2       2   
AMA18W8F60Y2J      24         24     24           24        24      24   
AVOF14300525D       1          1      1            1         1       1   

                CreationTime  MaxAssignments  RequesterAnnotation  \
WorkerId                                                            
A130P4M9JPT4YA             5               5                    5   
A1CE5G48ELWN0F             1               1                    1   
A1EUBMQ86K32XE             2               2                    2   
A1GKEEI844CEKI            17              17                   17   
A219VCQZADQ45W             2               2                    2   
A22VLXXVIEJSD1             1               1                    1   
A26UAJWMMNN351             1               1                    1   
A27SMEOPKV84VI             3               3                    3   
A28L1K6D8QUCML            12              12                   12   
A292QSLC0BUT0O             2               2                    2   
A29VL3MZE7YPBZ            12              12                   12   
A2IQ0QCTQ3KWLT             2               2                    2   
A2KLDW5VRXUJ8S             1               1                    1   
A2LRFPJOIC51KY            13              13                   13   
A2SBN9B6Q3DN2X             1               1                    1   
A2WNW8A4MOR7T7             9               9                    9   
A31Z5TPD8QKE26            25              25                   25   
A36H3PO6JZD414             6               6                    6   
A39YBLQYIUBR76            16              16                   16   
A3BI0AX5T5GVO3           129             129                  129   
A3HMDGRGC5JB7E             1               1           

In [11]:
batch_results_TEST_df['WorkTimeInSeconds'].mean()
# 7 minutes per task (?), quite a lot...
timings = batch_results_TEST_df[['AcceptTime', 'SubmitTime']]
timings.loc[:, 'AcceptTime'] = pd.to_datetime(timings['AcceptTime'])
timings.loc[:, 'SubmitTime'] =  pd.to_datetime(timings['SubmitTime'])
timings['diff'] = timings['SubmitTime'] - timings['AcceptTime']
diff = timings['diff']

/home/xilini/anaconda3/envs/vis-par/lib/python3.7/site-packages/dateutil/parser/_parser.py:1218: UnknownTimezoneWarning: tzname PDT identified but not understood.  Pass `tzinfos` argument in order to correctly return a timezone-aware datetime.  In a future version, this will raise an exception.
  category=UnknownTimezoneWarning)
/home/xilini/anaconda3/envs/vis-par/lib/python3.7/site-packages/pandas/core/indexing.py:1047: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[item_labels[indexer[info_axis]]] = value
/home/xilini/anaconda3/envs/vis-par/lib/python3.7/site-packages/pandas/core/indexing.py:965: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value in

In [12]:
diff.dt.total_seconds().min()

16.0

In [13]:
diff.dt.total_seconds().max()

1660.0